### PR TITLE
[catpowder] Bug fix: RSS queue detection issue

### PR DESCRIPTION
**Pull request by kyleholohan**
Change RSS queue count detection from using `RxRing`s to `TxRing`s. `RxRing`s do late-binding validation after the queue is programmed; `TxRing`s do immediate validation. This is causing issues with subsequent changes to catpowderW

### CODE REVIEW

File: src/rust/catpowder/win/runtime.rs [hunk 1]
```diff
@@ -228,6 +217,28 @@ fn count_processor_cores() -> Result<usize, Fail> {
...
+    let sys_proc_count: u32 = count_processor_cores()? as u32;
+
+    for queueid in 1..sys_proc_count {
+        match TxRing::new(api, DUMMY_QUEUE_LENGTH, ifindex, queueid) {
+            Ok(_) => (),
+            Err(e) => {
+                warn!(
+                    "Failed to create TX ring on queue {}: {:?}. This is only an error if {} is a valid RSS queue \
+                     ID",
+                    queueid, queueid, e
```
Comment (anandbonde) [inline src/rust/catpowder/win/runtime.rs]: Is this the correct ordering of the variables?

File: src/rust/catpowder/win/runtime.rs [hunk 2]
```diff
@@ -228,6 +217,28 @@ fn count_processor_cores() -> Result<usize, Fail> {
     return Ok(core_count);
 }
 
+/// Deduces the RSS settings for the given interface. Returns the number of valid RSS queues for the interface.
+fn deduce_rss_settings(api: &mut XdpApi, ifindex: u32) -> Result<u32, Fail> {
+    const DUMMY_QUEUE_LENGTH: u32 = 1;
+    let sys_proc_count: u32 = count_processor_cores()? as u32;
+
+    for queueid in 1..sys_proc_count {
```
Comment (anandbonde) [inline src/rust/catpowder/win/runtime.rs:227]: Why is this loop from 1 to sys_proc_count-1? It would be good to add a comment explaining why this is done.

**Review by anandbonde — APPROVED** [review]

File: src/rust/catpowder/win/runtime.rs (hunk 2 above)
Comment (kyleholohan) [reply src/rust/catpowder/win/runtime.rs:227]: Added a comment.

File: src/rust/catpowder/win/runtime.rs (hunk 1 above)
Comment (kyleholohan) [reply src/rust/catpowder/win/runtime.rs]: Good catch; thanks.

**Pull request merged**